### ISSUE TITLE
Test Monitor Issue

### ISSUE THREAD

**Issue by yl2chen** [livingsocial/rearview#45]
Hi Rearview Team,

Our team was on the Scala version and everything worked well.  Now we are setting up a new rearview instance using the latest version.

When I took an currently working Metric from one of our rearview monitors, and entered it on the new rearview instance, and hit test monitor, the following errors were shown.

Monitor Test Error!
Metrics contains an invalid metric: {working metrics I entered}

Server logs
I, [2014-06-19T22:06:14.118000 #433]  INFO -- : Started POST "/monitor.json" for 24.137.223.133 at 2014-06-19 22:06:14 +0000
I, [2014-06-19T22:06:14.156000 #433]  INFO -- : Processing by Rearview::MonitorController#create as JSON
I, [2014-06-19T22:06:14.157000 #433]  INFO -- :   Parameters: {"id"=>"", "dashboardId"=>"1", "userId"=>"2", "name"=>"test", "active"=>"true", "jobType"=>"monitor", "version"=>"2", "cronExpr"=>"0 0 \* ? Mar,May,Sep Sun,Mon,Wed,Thu,Fri,Sat", "errorTimeout"=>"60", "minutes"=>"15", "metrics"=>["{my metric}"], "monitorExpr"=>"limit = 25000\n\nif not @a.nil?\n  \nvalue = @a.values[-2] \n\nif not value.nil?\n\n\traise \"Max 95 Percentile for Get Anonymous Access Code is currently at #{value}) higher than the #{limit}\" if value > limit\n  \nend\n\nend", "toDate"=>"now", "createdAt"=>"", "modifiedAt"=>"", "description"=>""}
I, [2014-06-19T22:06:15.135000 #433]  INFO -- :   Rendered /home/ubuntu/.rvm/gems/jruby-1.7.5@rearview/gems/rearview-1.2.1-java/app/views/rearview/monitor/create.json.jbuilder (2.0ms)
I, [2014-06-19T22:06:15.137000 #433]  INFO -- : Completed 200 OK in 970ms (Views: 6.0ms | ActiveRecord: 2.0ms)

I have a valid hosted graphite connection as can be seen here https://github.com/livingsocial/rearview/issues/44

Any idea why its happening?

Thanks

**Comment by steveakers** [comment]
The latest rails version of rearview detects when a metric isn't in graphite and gives you that error. We added this functionality because we have a monthly purge of unused metrics and this was invalidating some monitors without us knowing. Have you checked to make sure all your metrics exist in graphite?

**Comment by yl2chen** [comment]
Yup, it is a metric that we currently monitor on using the scala version of rearview, and also I can get a valid graph with data, by hitting
https://www.hostedgraphite.com/{access_key}/graphite/render?target={this_metric}.

**Comment by steveakers** [comment]
Is there a whisper file for each of the metrics you're being told are invalid? (Sometimes graphite will create a graph if some of the metrics are valid.)

**Comment by yl2chen** [comment]
I was only testing with one single metric and I have verified it manually.
